implementations/rust/ockam/ockam_executor/src/channel.rs

```diff
@@ -104,39 +104,99 @@ impl<T> Drop for Sender<T> {
 
 impl<T> core::fmt::Debug for Sender<T> {
     fn fmt(&self, f: &mut core::fmt::Formatter) -> core::fmt::Result {
-        write!(f, "Sender")
+        write!(f, "[Sender]")
+    }
+}
+
+pub struct SendFuture<'a, T> {
+    inner: &'a Inner<T>,
+    value: Option<T>,
+}
+
+impl<'a, T> Future for SendFuture<'a, T>
+where
+    T: core::fmt::Debug,
+{
+    type Output = Result<(), error::SendError<T>>;
+
+    fn poll(mut self: Pin<&mut Self>, context: &mut Context<'_>) -> Poll<Self::Output> {
+        let value = self.value.take();
+        match value {
+            Some(value) => {
+                match self.inner.queue.enqueue(value) {
+                    Ok(()) => {
+                        self.inner.item_count.fetch_add(1, Ordering::Relaxed);
+                        self.inner.wake_receiver.wake();
+                        Poll::Ready(Ok(()))
+                    }
+                    Err(_) => {
+                        // queue is full - TODO implement backpressure
+                        {
+                            error!("queue overflowed");
+                            self.inner.is_sender_closed.swap(true, Ordering::AcqRel);
+                            self.inner.wake_receiver.wake();
+                        }
+                        self.inner.wake_sender.register(context.waker());
+                        Poll::Pending
+                    }
+                }
+            }
+            None => panic!("Value cannot be None"),
+        }
     }
 }
 
+impl<'a, T> Unpin for SendFuture<'a, T> {}
+
+/// Receiver
 pub struct Receiver<T>(Arc<Inner<T>>);
 
 impl<T: core::fmt::Debug> Receiver<T> {
```

Comment:
Why is there a `Debug` bound here?

Reply:
Whoops. That's left over from the early days to make it easier to debug the `no_std` implementation of `ockam::node`. I've removed them.